File: beadOrnaments.cpp
//My failed attempt at a c++ implementation
//Most likely does not work due to overflow errors
//Or something of that nature
//Works on test cases 0-5, but fails 6-9
#include <iostream>
using namespace std;

long long int myPow(long long int a, long long int b){
    if(b==0){
        return 1;
    }
    long long int ans=a;
    while(b>1){
        ans*=a;
        b--;
    }
    return ans;
}

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    long long int t,n,b;
    cin>>t;
    for(int i=0;i<t;i++){
        long long int ans=1,sum=0;
        cin>>n;
        for(int j=0;j<n;j++){
            cin>>b;
            sum+=b;
            ans*=myPow(b,b-1);
            ans = ans % 1000000007;
        }
        if(n-2>=0)
            ans*=myPow(sum,n-2);
        else
            ans/=sum;
        ans = ans % 1000000007;
        cout<<ans<<'\n';
    }
    return 0;
}

File: beadOrnaments.py
#I'm not gonna lie, I had to google this one
#as it took me a while to even understand what
#the question was asking for. I initially thought
#the ornament would be like a necklace and it took
#me a while to realize that the beads can be arranged
#as a tree, even though it explicitly said to think of it as a tree.
#Yeah, I ended up just implementing the code from the blog solution.
#I also tried to do it in C++, but I was failing half of the test
#cases most likely due to overflow errors.

import math

t = int(input())
for i in range(t):
    ans=1
    sumb=0
    n = int(input())
    str1 = raw_input()
    for b in str1.split():
        b=int(b)
        sumb+=b
        ans*=b**(b-1)
        ans = ans % 1000000007


    ans*=sumb**(n-2)
    ans = ans % 1000000007
    print int(ans)
